spec/shared_examples/after_commit.rb

```diff
@@ -0,0 +1,45 @@
+shared_context 'after_commit' do |options|
+  let(:transactions_count) { options[:transactions_count] }
+  let(:fail_transaction) { options[:fail_transaction] }
+
+  let(:transactions) { Array.new(transactions_count) { Neo4j::ActiveBase.new_transaction } }
+  let(:close_inner_transactions!) { transactions.reverse.first(transactions_count - 1).each(&:close) }
+  let(:to_or_not_to) { options[:fail_transaction] ? :not_to : :to }
+
+  it 'handles after_create_commit callbacks' do
+    company = Company.new
+
+    if transactions.empty?
+      expect { company.save }.to change { c.after_create_commit_called }
+    else
+      company.save
+      transactions.last.mark_failed if fail_transaction
+      close_inner_transactions!
+      expect { transactions.first.close }.send(to_or_not_to, change { c.after_create_commit_called })
+    end
+  end
+
+  it 'handles after_update_commit callbacks' do
+    c
```

Comment:
Having the `c` inherited from the scope that the shared context is used in was really confusing to me.  Can we have something like:

```ruby
shared_context 'after_commit' do |company_variable, options|
  let(:company) { send(:company_variable) }
end
```